### PR TITLE
magento/devdocs#: Layout instructions. Adding more examples to Public interfaces & APIs [GITBUG]

### PR DESCRIPTION
**Pull request by DenisSaltanahmedov**
## Purpose of this pull request

This pull request (PR) adds code examples to the Public interfaces and APIs topic, I think the code examples will be helpful for developers

## Affected DevDocs pages
-  https://devdocs.magento.com/guides/v2.3/extension-dev-guide/api-concepts.html

## Links to Magento source code
-  https://github.com/magento/devdocs/blob/master/src/guides/v2.3/extension-dev-guide/api-concepts.md

whatsnew
Added the 'Example of an API interface implementation' section to the [Public interfaces & APIs](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/api-concepts.html) topic.

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: src/guides/v2.3/extension-dev-guide/api-concepts.md [hunk 1]
```diff
@@ -15,10 +15,84 @@ Public interfaces for a [module](https://glossary.magento.com/module) are marked
...
+interface CatalogRuleRepositoryInterface
+{
+...
+```
+
 ## What is an API? {#api-definition}
 
 An application programming interface (API) is a set of interfaces and their implementations that a module provides to other modules.
 
+### Example of implementation a set of API interfaces
```
Comment (rogyar) [inline src/guides/v2.3/extension-dev-guide/api-concepts.md]: Hi @DenisSaltanahmedov. I would suggest shrinking this example to just one interface declaration and the corresponding `di.xml` file. Also, I would add extra information about an interface implementation. Like "a default implementation for the interface may be declared in the following way ... ".  

So, just a little more information on what is the examples about. 

Thank you!

**Comment by DenisSaltanahmedov** [comment]
Hi @rogyar, thanks for your review.
I have fixed the recommendations. And added the example of an interface implementation
Thanks.

File: src/guides/v2.3/extension-dev-guide/api-concepts.md [hunk 2]
```diff
@@ -15,10 +15,134 @@ Public interfaces for a [module](https://glossary.magento.com/module) are marked
...
+interface CatalogRuleRepositoryInterface
+{
+...
+```
+
 ## What is an API? {#api-definition}
 
 An application programming interface (API) is a set of interfaces and their implementations that a module provides to other modules.
 
+### Example of implementation a API interface
```
Comment (rogyar) [inline src/guides/v2.3/extension-dev-guide/api-concepts.md]:
```suggestion
### Example of an API interface implementation
```

File: src/guides/v2.3/extension-dev-guide/api-concepts.md [hunk 3]
```diff
@@ -15,10 +15,134 @@ Public interfaces for a [module](https://glossary.magento.com/module) are marked
...
+     * @param int $ruleId
+     * @return bool
+     * @throws \Magento\Framework\Exception\CouldNotDeleteException
+     * @since 100.1.0
+     */
+    public function deleteById($ruleId);
+}
+```
+
+The implementations of the interfaces in the ``di.xml``
```
Comment (rogyar) [inline src/guides/v2.3/extension-dev-guide/api-concepts.md]:
```suggestion
The implementation of each interface is declared in the ``di.xml``
```

**Comment by DenisSaltanahmedov** [comment]
Hi @rogyar, fixed.
Thanks.

**Comment by DenisSaltanahmedov** [comment]
Hi @dobooth, any updates about the issue? Thanks.

**Comment by jeff-matthews** [comment]
@DenisSaltanahmedov, the **Update branch** button isn't working. We've opened a ticket with GitHub support to fix it. If we can't get support to help us by next week, we'll start using workarounds to merge all affected PRs.

**Comment by DenisSaltanahmedov** [comment]
Hi, if we still have the problem with merge in the PR, I can recreate the changes to new PR from the current master branch. Thanks.

**Comment by dshevtsov** [comment]
@DenisSaltanahmedov Looks like the issue was fixed finally. Thank you for your patience.

**Comment by dobooth** [comment]
running tests

**Pull request merged**